tutorial/part-1-creating-a-simple-view/creating-your-first-table.md

```diff
@@ -0,0 +1,141 @@
+
+1. The first step is to set up the database query that fetches this data.
+2. The second step is to configure the table to show the query's results.
+
+## Writing your first query
+
+1. Click on **+** next to **DB Queries**
+2. Choose **Mock Database**
+3. Note that the **Mock Database** is a PostgreSQL database
+4. Click on **New Query**
+5. You’ll see a query created with the name **Query1**
+6. Rename the query to **ProductsQuery**
+7. Copy-paste the query shown below in the **Query** tab `SELECT "productId", "productName", "category", "mrp" FROM products ORDER BY "productId";`
+8. Click on **Run**
+9. Note the pop-up informing that the query ran successfully
+
+{% hint style="info" %}
+**Mock Database:**
+
+Your Personal Organization comes with a pre-configured mock PostgreSQL Database called Mock Database. However, when you build your own app, you’ll connect to your own database. If you’re new to Appsmith, read [this](https://docs.appsmith.com/core-concepts/connecting-to-databases) to learn to configure a database of your choice.
+{% endhint %}
+
+## Accessing query results from a widget
+
+The next step is to display the query results in the **Products\_Table** :
+
+1. Navigate to **ProductListPage → Widgets → Products\_Table**
+2. Open **Product\_Table’**s properties
+3. Type the following in the Table Data property: `{{ProductsQuery.data}}`
+4. Your table now displays ProductsQuery’s results
+5. Click the Deploy button on the top right
+6. Click on the down arrow button next to the deploy button to get the URL of your app
+
+Visit the app URL with your web browser. You'll see a page displaying your query results in a table like below:  
+
+
+![](https://lh5.googleusercontent.com/wjbhU2Nsq_tfEFoAsI4qEn60jo6E8dkySMMUqoV9h1IdfBJ9Ug48_EkI-LZVaRK3VB4ebTi0OTbYFczticODH13A-XWJi-qhE12Lhz8OSXnCvRDB6uqceArq3wDVZA5xOaQlAogK)
+
+Let’s see how this works:
+
+1. By using mustaches, you’re asking Appsmith to resolve anything within it as JavaScript code.
+2. By using the query’s name within mustaches `{{ ProductsQuery }}`, you’re accessing the entire query object of **ProductsQuery**.
+3. By calling the data function on **ProductsQuery**, you’re accessing its results. 
+4. By setting the **Table Data** to `{{ ProductsQuery.data }}`, you’re telling **Products\_Table** to display the results of **ProductsQuery**.
+
+Note that you are accessing a query's result in a widget by calling a method on the query's name. As you'll see in the next section, the inverse of this is also possible, i.e a widget's state can also be accessed by a query. Furthermore, all the building blocks of an Appsmith page - Widgets, DB Queries, and APIs can access each other's data and/or state using their names. 
+
+{% hint style="info" %}
+Think of widgets, APIs and DB Queries in Appsmith as variables in your programming language. Similar to variables:
+
+* They are identified by a name
```

Comment:
I see the general style is to give the instructions and then explain ourselves a little later in the docs. I'm not sure this sits well with people in general. What do you think?

Reply:
The reason for choosing this is to first make the user do something that shows them that it works; and then help them understand how it works. It's similar to the structure of most programming tutorials - help the reader print hello world, then explain everything.